Helper/Price.php

```diff
@@ -192,20 +192,49 @@ private function getConfigurableProductPrice($product, $type)
      */
     private function getMinimumComplexProductPrice($product, $usedProds, $type)
     {
-        $prices = [];
+        $minimum_price = null;
+        $minimum_variant = null;
+
+        /*
+        We identify the variant with the minimum final price (or price), and
+        that is the one that is used to obtain the requested price for the
+        parent as the variant is chosen as the representative for the product
+        */
         foreach ($usedProds as $child) {
             if ($child->getId() != $product->getId()) {
-                $price = $child->getPriceInfo()->getPrice($type);
-                $prices['prices'][] =  $price;
-                $prices['values'][] =  $price->getAmount()->getValue();
+                $variant_minimum_price = $this->getMinimumVariantPrice($child);
+
+                if ($minimum_price === null) {
+                    $minimum_price = $variant_minimum_price;
+                    $minimum_variant = $child;
+                } elseif ($variant_minimum_price < $minimum_price) {
+                    $minimum_price = $variant_minimum_price;
+                    $minimum_variant = $child;
+                }
             }
         }
 
-        if (empty($prices)) {
+        if (is_null($minimum_variant)) {
```

Comment:
Why are we using here `is_null` and `=== null` in other places?
@sofia-doofinder  Which is preferred?

Reply:
Agree with both suggestions! Including both :)